core/sail/nativerdf/src/main/java/org/eclipse/rdf4j/sail/nativerdf/ContextStore.java

```diff
@@ -100,8 +101,13 @@
 
 		try {
 			readContextsFromFile();
-		} catch (IOException e) {
-			logger.info("could not read context index: " + e.getMessage(), e);
+		} catch (FileNotFoundException fe) {
+			logger.info("context index has not been created yet: " + fe.getMessage());
+			initializeContextCache();
+			writeContextsToFile();
+			logger.info("context index construction complete");
```

Comment:
Propose .debug(...) since the user usually doesn't need to know about this.

Reply:
good idea
FWIW, this PR also reduces the noise in the output of the unit tests, since quite a few NativeStores are created in temp directories